### DISCUSSION TITLE
How to group updates by component and path in a generic way

### DISCUSSION THREAD

**Discussion by jkotiuk** [renovatebot/renovate#33069] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub

### Please tell us more about your question or problem

I manage a GitOps repo with the following directory structure:

```
apps/
└── component1/
    ├── base/
    └── overlays/
        ├── local/
        ├── preprod/
        └── prod/
└── componentN/
    ├── base/
    └── overlays/
        ├── local/
        ├── preprod/
        └── prod/
```

I'm trying to configure Renovate to:

Dynamically group updates by component (e.g., component1, componentN), without defining each package or component individually as packageRule:
* Combine updates for `overlays/local` and `overlays/preprod` into a single PR per component.
* Create separate PRs for updates in base, prod overlay inherits the version from it, so it should have its own PR.

This should work generically, automatically handling new components without additional configuration. i.e something as pseudo code below
```
  "packageRules": [
    {
      "description": "Group local and preprod overlay updates per component",
      "matchPaths": ["**/overlays/local/**", "**/overlays/preprod/**"],
      "groupName": "Overlays",
      "packageName": "{{packageName}}" // Group the same packages in a dynamic way
    }
  ]
  ```
  
  Is that possible with current Renovate release?

**Comment by rarkins** [comment, marked as answer]
Never change packageName like that. Use groupName for that - that's its purpose. Eg groupName: "{{parentDir}} dependencies"

**Reply by jkotiuk** [reply]
I'm not changing that, just wondering if there is a generic way to group packages by path.

As an example below config would work but I'd have to define 100+ rules, each for one packageName, on top of that I would have to add new rule every time I add a new package to the repository. On the other hand, if I do not provide matchPackageNames it will create one big PR for all packages in the repository.
 
```
  "packageRules": [
    {
      "description": "Group local and preprod overlay updates per component",
      "matchPaths": ["**/overlays/local/**", "**/overlays/preprod/**"],
      "groupName": "Overlays",
      "matchPackageNames": ["redis"]
    }
  ]
 ```

**Reply by rarkins** [reply]
I gave you a suggestion for doing that above

**Reply by jkotiuk** [reply]
Figured it out, thanks.

I've used `"groupName": "{{packageName}} Overlays"`. Initialy it didn't work because I had set global `additionalBranchPrefix` as part of testing.

**Reply by seanturner026** [reply]
@jkotiuk sorry to bother you but what does your config look like exactly if you don't mind?

I've got this sort of layout which is super similar (dependency declared in kustomization.yaml). 

```
.
└── apps
    ├── argo-rollouts
    │   ├── overlays
    │   │   ├── afs1-production
    │   │   │   └── kustomization.yaml
    │   │   ├── usw2-dev
    │   │   │   └── kustomization.yaml
    │   │   ├── usw2-production
    │   │   │   └── kustomization.yaml
    │   │   └──usw2-staging
    │   │       └── kustomization.yaml
   ... and so on
```

My config is causing a problem where any PR opened will be changing all environments which doesn't quite work.

This doesn't seem to work with `{{packageName}}` or `{{parentDir}}` :(

```json
{
	"extends": [
		"config:recommended"
	],
	"packageRules": [
		{
			"matchFileNames": [
				"apps/.+/overlays/afs1-production/.+/kustomization\\.yaml$"
			],
			"groupName": "afs1-production {{parentDir}} Overlays",
			"labels": ["afs1-production"]
		},
		{
			"matchFileNames": [
				"apps/.+/overlays/usw2-dev/.+/kustomization\\.yaml$"
			],
			"groupName": "usw2-dev {{parentDir}} Overlays",
			"labels": ["usw2-dev"]
		},
		{
			"matchFileNames": [
				"apps/.+/overlays/usw2-production/.+/kustomization\\.yaml$"
			],
			"groupName": "usw2-production {{parentDir}} Overlays",
			"labels": ["usw2-production"]
		}
	]
}
```
